sim/vcf_sample.py

```diff
@@ -0,0 +1,81 @@
+import math
+import os
+import errno
+
+from utils import run
+
+from sample import Sample
+from genome import SimulatedGenome
+import config
+
+class VcfSample(Sample):
+    """ A sample generated from vcf file using simuG """
+    def __init__(self, 
+        predef_snp_path, 
+        reference_path=config.DEFAULT_REFERENCE_PATH,
+        seed=1, 
+        per_base_error_rate="0",
+        num_read_pairs=config.DEFAULT_NUM_READ_PAIRS,
+    ):
+
+        # TODO: Validate
+        self.reference_path = reference_path
+        
+        self.seed = seed
+        self.per_base_error_rate = per_base_error_rate
+        self.num_read_pairs = math.ceil(num_read_pairs)
+        
+        if os.path.isfile(predef_snp_path):
+            self.predef_snp_path = predef_snp_path
+        else:
+            raise Exception("Cant find the pre-defined snp path")
+        
+    @property
+    def name(self):
+        # extract filename
+        basename = os.path.splitext(os.path.basename(self.predef_snp_path))[0]
+        #TODO: this pattern is getting somewhat unweildy.
+        #   etiher use something more general or save metadata separately
+        return f"{type(self).__name__}-{basename}-seed{self.seed}-error{self.per_base_error_rate}-readpairs-{self.num_read_pairs}"
+
+    def decompose_complex_snps(self, output_file_path):
+        run(['vt', 
+            'decompose_blocksub',
+            self.predef_snp_path,
+            '-o', output_file_path])
+
+    def simulate_genome(self, simulated_genome_path, seed=1):
+        """ Simulated a genome with random SNPs
```

Comment:
Should be `""" Simulate genome with predefined SNPs...`

Reply:
just changed it to something more general. The class description  covers this detail